pkg/controller/operconfig/operconfig_controller.go

```diff
@@ -267,8 +268,19 @@ func (r *ReconcileOperConfig) Reconcile(ctx context.Context, request reconcile.R
 		}
 	}
 
+	// If we use hypershift, the components need proxy settings. In selfhosted, they
+	// are always local to the API so there it is never needed.
+	var proxyConfig configv1.ProxyStatus
+	if hyperShiftConfig := network.NewHyperShiftConfig(); hyperShiftConfig.Enabled {
```

Comment:
Technically this shouldn't be needed btw, as in selfhosted the KAS internal address is in `NO_PROXY`

Reply:
We tend to gather required settings via the Bootstrap mechanism -- can you move this to pkg/platform/platform.go? (And update the InfraStatus type accordingly?)

Reply:
@squeed updated, PTAL